src/server/sass/build_all.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Build } from './build';
+
+export function buildAll(enabledPlugins, { onSuccess, onError }) {
+  return Promise.all(enabledPlugins.reduce((acc, plugin) => {
+    if (!plugin.getExportAppStyleSheetToCompile()) {
```

Comment:
It would be ideal if this used `collectUiExports()` from `src/ui/ui_exports` instead of adding methods for reading the uiExports to the plugin spec. `collectUiExports()` knows how to merge the uiExports from all plugins and will normalize the specs, applying defaults and failing if the specs are invalid. It would probably look like this:

```js
const { uiAppSpecs = [] } = collectUiExports(enabledPluginSpecs);
return Promise.all(uiAppSpecs.map(async uiAppSpec => {
  if (!uiAppSpec.styleSheetPath) {
    return
  }

  const builder = new Build(uiAppSpec.styleSheetPath, {
    onSuccess,
    onError,
  });
 
  await builder.build();
}));
```